cmd/kubelet/kubelet.go

```diff
@@ -93,9 +93,10 @@ func main() {
 
 	// use kubeletServer to construct the default KubeletDeps
 	kubeletDeps, err := app.UnsecuredDependencies(kubeletServer)
-
-	// add the kubelet config controller to kubeletDeps
-	kubeletDeps.KubeletConfigController = kubeletConfigController
+	if err == nil {
```

Comment:
I don't think kubelet can function without the dependencies. If the function returns an error, kubelet should probably just die.